### PR TITLE
base_admin_controller hostname refactor

### PR DESCRIPTION
**Pull request by btidwell**

### CODE REVIEW

File: plugins/pencilblue/controllers/admin/content/articles/article_form.js [hunk 1]
```diff
@@ -137,8 +137,7 @@ module.exports = function(pb) {
             media: data.media,
             article: data.article,
             siteKey: pb.SiteService.SITE_FIELD,
-            site: self.site,
-            sitePrefix: self.sitePrefix
+            site: self.site
```
Comment (andparker) [inline plugins/pencilblue/controllers/admin/content/articles/article_form.js]: We might be able to remove site here entirely?

File: plugins/pencilblue/controllers/admin/content/pages/page_form.js [hunk 2]
```diff
@@ -123,8 +123,7 @@ module.exports = function(pb) {
             media: data.media,
             page: data.page,
             siteKey: pb.SiteService.SITE_FIELD,
-            site: self.site,
-            sitePrefix: self.sitePrefix
+            site: self.site
```
Comment (andparker) [inline plugins/pencilblue/controllers/admin/content/pages/page_form.js]: Might be able to remove site here, too?

File: plugins/pencilblue/controllers/admin/content/articles/article_form.js (hunk 1 above)
Comment (btidwell) [inline plugins/pencilblue/controllers/admin/content/articles/article_form.js]: Can't remove it.  It's used in the angular controller for the url availability api.  This would have to be changed in another pr/card.  The api would have to resolve using hostname and the angular controller would have to be updated.   The site value is passed in as a query and not a path var.
